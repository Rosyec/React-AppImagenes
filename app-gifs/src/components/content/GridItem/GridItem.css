.item {
    box-shadow: 0px 0px 50px 0px rgba(0,0,0,0.1);
}
.img-fluid {
    min-width: 300px;
}