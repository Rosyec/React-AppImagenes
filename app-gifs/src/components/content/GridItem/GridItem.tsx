import "./GridItem.css";

export const GridItem = ({ url, alt_description }: Properties) => {
  return (
    <div className="col-md-5 col-sm-12 item m-1 d-flex justify-content-center align-items-center g-0 animate__animated animate__fadeIn">
      <img src={url} alt={alt_description} className="img-fluid" />
    </div>
  );
};

interface Properties {
  url: string;
  alt_description: string;
}
