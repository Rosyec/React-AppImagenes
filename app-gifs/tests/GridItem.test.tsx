import { render, screen } from "@testing-library/react";
import React from "react";
import { GridItem } from "../src/components/content/GridItem/GridItem";


describe('Pruebas <GridItem/>', () => {

    const url: string = 'http://www.dragon/dragon.jpg';
    const alt: string = 'Dragon'

    test('Las props url y alt_description sean obligatorias.', () => {
        const { container } = render(<GridItem url={url} alt_description={alt}></GridItem>);
        expect(container).toMatchSnapshot();
    });

    test('Debe de mostrar una imagen', () => {
        render(<GridItem url={url} alt_description={alt}></GridItem>);
        expect(screen.getByRole('img')).toBeTruthy();
    });

    test('Debe de mostrar una imagen con el url y el alt_description indicado', () => {
        render(<GridItem url={url} alt_description={alt}></GridItem>);
        expect(screen.getByRole('img').getAttribute('src')).toBe(url);
        expect(screen.getByRole('img').getAttribute('alt')).toBe(alt);
    });


});


export { }