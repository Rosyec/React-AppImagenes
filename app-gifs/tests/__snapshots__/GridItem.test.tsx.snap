// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas <GridItem/> Las props url y alt_description sean obligatorias. 1`] = `
<div>
  <div
    class="col-md-5 col-sm-12 item m-1 d-flex justify-content-center align-items-center g-0"
  >
    <img
      alt="Dragon"
      class="img-fluid"
      src="http://www.dragon/dragon.jpg"
    />
  </div>
</div>
`;
